lansa-vmss-windows-autoscale-sql-database/azuredeploy.json

```diff
@@ -1,56 +1,154 @@
 {
-   "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+   "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
-      "msiUri": {
+      "location": {
+         "type": "string",
+         "defaultValue": "[resourceGroup().location]",
+         "metadata": {
+            "description": "Resource Group Location."
+         }
+      },
+      "msiURL": {
          "type": "string",
          "defaultValue": "https://lansalpcmsdn.blob.core.windows.net/app/test/AWAMAPP_v14.1.2_en-us.msi",
          "metadata": {
-            "description": "URL of the LANSA MSI to install on each VM."
+            "description": "URL of the LANSA MSI which will be installed on each virtual machine."
          }
       },
-      "vmSku": {
+      "virtualMachineSize": {
          "type": "string",
-         "defaultValue": "Standard_A2",
+         "defaultValue": "Standard_B4ms",
          "metadata": {
-            "description": "Size of VMs in the VM Scale Set."
+            "description": "Size of the Virtual Machines in the Virtual Machine Scale Set."
          }
       },
-      "vmssName": {
+      "stackName": {
          "type": "string",
          "metadata": {
-            "description": "String used as a base for naming resources. Must be 3-61 characters in length and globally unique across Azure. A hash is prepended to this string for some resources, and resource-specific information is appended."
+            "description": "String used as a base for naming resources. Must be 3-61 characters in length and globally unique across Azure. A hash is prepended to this string for some resources, and resource-specific information is appended. Some identifiers use precisely 9 characters from this name and so it can be useful to use exactly 9. The template pads it out or truncates it as necessary to make it 9 characters long where required."
          },
+         "minLength" : 3,
          "maxLength": 61
```

Comment:
if this needs to be 9 chars, set the maxLength to 9

Reply:
It doesn't need to be 9 chars. Its just that some usage of this value truncates it to 9 chars. Hence the description. Why its truncated to 9 chars is historical and I'd like to change it but it will break customer's deployments who are currently using the template, when they upgrade to using it. Like the DNS name will change.